bundle/edu.gemini.pot/src/test/scala/edu/gemini/spModel/target/env/GuideGroupSpec.scala

```diff
@@ -177,90 +175,92 @@ class GuideGroupSpec extends Specification with ScalaCheck with Arbitraries {
       }
 
     "return matching guide probe targets for all guide probes with associated targets" in
-      forAll { (g: GuideGroup) =>
+      forAll { g: GuideGroup =>
         val gpts1 = g.getAll.asScalaList
         val gpts2 = AllProbes.flatMap { gp => g.get(gp).asScalaOpt }
         gpts1 == gpts2
       }
   }
-/*
-  "GuideGroup getAll" should {
-    "" in
-      forAll { (g: GuideGroup) =>
-        true
-      }
-
-    "" in
-      forAll { (g: GuideGroup) =>
-        true
-      }
-
-    "" in
-      forAll { (g: GuideGroup) =>
-        true
-      }
-
-    "" in
-      forAll { (g: GuideGroup) =>
-        true
-      }
-
-    "" in
-      forAll { (g: GuideGroup) =>
-        true
-      }
 
-    "" in
-      forAll { (g: GuideGroup) =>
-        true
+  "GuideGroup putAll" should {
+    "contain the original primary guide probes and all primary guide probes for which targets have been added" in
+      forAll { (g: GuideGroup, gps: Set[GuideProbe]) =>
+        val gpts = gps.map(GuideProbeTargets.create(_, new SPTarget())).toList.asImList
+        val allGps = g.getPrimaryReferencedGuiders.asScala.toSet ++ gps
+        g.putAll(gpts).getPrimaryReferencedGuiders.asScala.toSet == allGps
       }
 
-    "" in
-      forAll { (g: GuideGroup) =>
-        true
-      }
   }
 
-  "GuideGroup getAll" should {
-    "" in
-      forAll { (g: GuideGroup) =>
-        true
+  "GuideGroup setAll" should {
+    "only contain the guide probes in the collection" in
+      forAll { (g: GuideGroup, gps: Set[GuideProbe]) =>
+        val gpts = gps.map(GuideProbeTargets.create(_, new SPTarget())).toList.asImList
+        val g2 = g.setAll(gpts)
+        val gpOut = g.getReferencedGuiders.asScala.toSet -- g2.getReferencedGuiders.asScala.toSet
+        g2.getReferencedGuiders.asScala.toSet == gpts.asScalaList.map(_.getGuider).toSet &&
+          gpOut.forall(gp => !g2.contains(gp))
```

Comment:
Here too we should add a test that verifies that it is the same as calling `GuideGroup.clear` and then calling `GuideGroup.putAll`.  

Also, I guess we could add an `Arbitraries` entry for `ImList` and one for `GuideProbeTargets` so they don't need to be explicitly created.